graas/api.py

```diff
@@ -1,32 +1,37 @@
 """ GRaaS API. """
 
 import pkg_resources
-
 from klein import Klein
-
-from twisted.python.filepath import FilePath
 from twisted.web.static import File
 
+from .views import HomePageElement
 
-class GraasApi(object):
 
+class GraasApi(object):
     app = Klein()
     static_path = pkg_resources.resource_filename(__name__, "static")
 
     def __init__(self, gate_remote):
         self._gate_remote = gate_remote
 
-    @app.route('/')
+    @app.route('/', methods=['POST', 'GET'])
     def home(self, request):
-        p = FilePath(self.static_path).child("index.html")
-        return p.open().read()
+        message = ''
+        message_type = ''
+        pin = request.args.get('gatecode', [''])[0]
+
+        if self._gate_remote and pin:
+            if str(pin) == str(self._gate_remote._gate_remote_pin):
```

Comment:
`gate_remote_pin` is not the passphrase here, it's the physical pin number that the remote is connected to.

We will need to add another config option here for the code, and maybe rename the `pin` variable in this method to avoid confusion.